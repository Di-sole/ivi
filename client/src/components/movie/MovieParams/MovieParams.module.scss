.params {
    width: 100%;

    &__list {
        display: flex;
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
        align-items: center;
        justify-content: center;
        color: rgba(255, 255, 255, 0.8);

        @media (max-width: 1159px) {
            justify-content: start;
        }
        @media (max-width: 879px) {
            justify-content: center;
        }

        &Item {
            position: relative;
            font-size: 15px;
            line-height: 20px;
            font-weight: 400;
            display: inline-flex;
            vertical-align: top;

            &:not(:first-child)::before {
                content: '';
                width: 8px;
                display: inline-block;
            }

            &_point {
                display: inline-flex;
                align-items: center;
            }
            &_point::before {
                content: '';
                text-align: center;
            }

            &_point:not(:first-child)::before {
                width: 2px;
                height: 2px;
                background-color: rgba(255, 255, 255, 0.72);
                margin-left: 6px;
                margin-right: 6px;
                margin-bottom: 3px;
            }
        }

        &Icon {
            width: 15px;
            height: 15px;
            margin-left: 6px;
            margin-right: -5px;
        }

        &:not(:first-child) {
            margin-top: 4px;
        }
    }
}
