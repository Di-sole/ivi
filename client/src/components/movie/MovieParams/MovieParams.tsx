import styles from './MovieParams.module.scss';
import { IMovie } from '../movieMedallion/MovieMedallionsList/Temp/IMovie';
import TextLinkUI from '../../UI/links/TextLink/TextLinkUI';
import TextBadge from '../../UI/badges/TextBadge/TextBadge';
import Image from 'next/image';
import soundIcon from '../../../../public/icons/sound.png';
import subtitlesIcon from '../../../../public/icons/subtitles.png';

interface MovieParamsProps {
    movie: IMovie;
}

const MovieParams = ({ movie }: MovieParamsProps) => {
    return (
        <div className={styles.container} data-testid="div-params">
            <div className={styles.container__list}>
                <TextLinkUI href="/" option="bright">
                    {movie.year}
                </TextLinkUI>
                <span className={styles.container__listItem}>{movie.time}</span>
                <span className={styles.container__listItem}>{movie.ageLimit}</span>
            </div>
            <div className={styles.container__list}>
                <TextLinkUI
                    href="/"
                    option="bright"
                    className={[styles.container__listItem, styles.container__listItem_point].join(' ')}
                >
                    {movie.production}
                </TextLinkUI>
                {movie.genres.slice(0, 3).map((genre) => (
                    <TextLinkUI
                        key={genre}
                        href="/"
                        option="bright"
                        className={[styles.container__listItem, styles.container__listItem_point].join(' ')}
                    >
                        {genre}
                    </TextLinkUI>
                ))}
            </div>
            <div className={styles.container__list}>
                <TextBadge text={movie.displays[0]} />
                <Image
                    src={soundIcon}
                    alt="soundIcon"
                    className={[styles.container__listIcon, styles.container__listIcon_sound].join(' ')}
                />
                <div className={styles.container__listItem}>
                    {movie.voiceActing.map((voice) => (
                        <div
                            key={voice}
                            className={[styles.container__listItem, styles.container__listItem_point].join(' ')}
                        >
                            {voice}
                        </div>
                    ))}
                </div>
                {movie.subtitles.length > 0 && (
                    <>
                        <Image src={subtitlesIcon} alt="subtitlesIcon" className={styles.container__listIcon} />
                        <div className={styles.container__listItem}>
                            {movie.subtitles.map((sub) => (
                                <div
                                    key={sub}
                                    className={[styles.container__listItem, styles.container__listItem_point].join(' ')}
                                >
                                    {sub}
                                </div>
                            ))}
                        </div>
                    </>
                )}
            </div>
        </div>
    );
};

export default MovieParams;
