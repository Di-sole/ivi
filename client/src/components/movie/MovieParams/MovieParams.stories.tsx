import { Meta, StoryObj } from '@storybook/react';
import MovieParams from './MovieParams';
import '../../../styles/nullstyle.scss';
import '../../../styles/globals.scss';
import './MovieParams.module.scss';
import { IMovie } from '../../squareCard/SquareCardsList/Temp/IMovie';

const meta: Meta<typeof MovieParams> = {
    title: 'Information/Params',
    component: MovieParams,
};

export default meta;

type Story = StoryObj<typeof MovieParams>;

const data: IMovie = {
    id: 1,
    title: 'Название',
    year: 1900,
    ageLimit: '0+',
    time: 'Длительность',
    production: 'Производство',
    genres: ['Жанр', 'Жанр', 'Жанр'],
    displays: ['FullHD', 'HD', '1080', '720'],
    voiceActing: ['Рус', 'Eng'],
    subtitles: ['Рус'],
    subtitlesFull: ['Русский'],
    description: [],
    language: ['Русский', 'Английский'],
    raiting: 9.5,
    nameraiting: 'Рейтинг',
    actors: [],
};

export const FullScreen: Story = {
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19' }}>
                <Story />
            </div>
        ),
    ],
    args: {
        movie: data,
    },
};
