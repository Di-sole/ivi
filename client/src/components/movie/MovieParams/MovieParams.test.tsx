import { render, screen } from '@testing-library/react';
import MovieParams from './MovieParams';
import { movies } from '../movieMedallion/MovieMedallionsList/Temp/Movie.data';

describe('MOVIEPARAMS SNAPSHOTS TESTS', () => {
    test('MovieParams should not be changed', () => {
        render(<MovieParams movie={movies[0]} />);

        const div = screen.getByTestId('div-params');
        expect(div).toMatchSnapshot();
    });
});
