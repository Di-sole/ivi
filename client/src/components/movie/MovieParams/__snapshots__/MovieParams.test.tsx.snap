// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOVIEPARAMS SNAPSHOTS TESTS MovieParams should not be changed 1`] = `
<div
  class="container"
  data-testid="div-params"
>
  <div
    class="container__list"
  >
    <a
      class="container container_bright"
      href="/"
    >
      2010
    </a>
    <span
      class="container__listItem"
    >
      2 ч. 14 мин.
    </span>
    <span
      class="container__listItem"
    >
      6+
    </span>
  </div>
  <div
    class="container__list"
  >
    <a
      class="container container_bright container__listItem container__listItem_point"
      href="/"
    >
      США
    </a>
    <a
      class="container container_bright container__listItem container__listItem_point"
      href="/"
    >
      Драмы
    </a>
    <a
      class="container container_bright container__listItem container__listItem_point"
      href="/"
    >
      Спорт
    </a>
    <a
      class="container container_bright container__listItem container__listItem_point"
      href="/"
    >
      Семейные
    </a>
  </div>
  <div
    class="container__list"
  >
    <div
      class="badge"
    >
      FullHD
    </div>
    <img
      alt="soundIcon"
      class="container__listIcon container__listIcon_sound"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <div
      class="container__listItem"
    >
      <div
        class="container__listItem container__listItem_point"
      >
        Рус
      </div>
      <div
        class="container__listItem container__listItem_point"
      >
        Eng
      </div>
    </div>
    <img
      alt="subtitlesIcon"
      class="container__listIcon"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <div
      class="container__listItem"
    >
      <div
        class="container__listItem container__listItem_point"
      >
        Рус
      </div>
    </div>
  </div>
</div>
`;
